### PR TITLE
Support loading /pr/kubernetes/kubernetes/1234 correctly

### PR DESCRIPTION
**Pull request by fejta**
/assign @fejta

Support all the following:
```
/pr/1234 == /pr/kubernetes/kubernetes/1234 == /pr/kubernetes/1234
/pr/test-infra/1234 == /pr/kubernetes/test-infra/1234
/pr/google_cadvisor/1234 == /pr/google/cadvisor/1234
```
(add unit tests for this behavior)

Always display the org/repo#1234 on the pr result page

Correctly link to the google/cadvisor repo

ref https://github.com/kubernetes/test-infra/issues/920

### CODE REVIEW

**Comment by rmmh** [comment]
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[gubernator/app.yaml, line 2 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUASU6KLfA7rojL2Ds:-KqUASU6KLfA7rojL2Dt:b9tqxnt) ([raw file](https://github.com/kubernetes/test-infra/blob/261965b5015d11db73a1a9b5df95ca2077316982/gubernator/app.yaml#L2)):*
> ```YAML
> application: k8s-gubernator
> version: 1
> ```

why is this changing?

---

*[gubernator/view_pr.py, line 34 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUAdl8LH-Y0bFbS5KS:-KqUAdl8LH-Y0bFbS5KT:bc8mir8) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr.py#L34)):*
> ```Python
> @view_base.memcache_memoize('pr-details://', expires=60 * 3)
> def pr_builds(path):
>     """Return {job: [(build_number, {started.json}, {finished.json})]} for the PR path."""
> ```

`for each job under a PR path`

---

*[gubernator/view_pr.py, line 81 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUB6AOUsjI8tjvr33X:-KqUB6AOUsjI8tjvr33Y:b-62mes2) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr.py#L81)):*
> ```Python
>     # https://k8s-gubernator.appspot.com/pr/google_cadvisor/1707
>     # https://k8s-gubernator.appspot.com/pr/44245
>     # https://k8s-gubernator.appspot.com/pr/test-infra/3792
> ```

are these comments necessary?

---

*[gubernator/view_pr.py, line 114 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUBSQ1AyjADpdn_ntr:-KqUBSQ1AyjADpdn_nts:bycrhin) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr.py#L114)):*
> ```Python
> 
>         max_builds, headings, rows = pull_request.builds_to_table(builds)
>         digest = ghm.GHIssueDigest.get('/'.join([org, repo]), pr)
> ```

`'%s/%s' % (org, repo)` ?

---

*[gubernator/view_pr_test.py, line 42 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqU9wRofPyHcBugkN5f:-KqU9wRofPyHcBugkN5g:bp7fw8r) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr_test.py#L42)):*
> ```Python
>     ('/google/cadvisor', 'google', 'cadvisor'),
> ])
> def test_org_repo(path, org, repo):
> ```

Please match the format of the other Gubernator tests. See regex_test.py for an example of how this might look.

---

*[gubernator/templates/pr.html, line 4 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUCnvqUur5BqQJXspX:-KqUCnvqUur5BqQJXspY:bi2gg10) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/templates/pr.html#L4)):*
> ```HTML
> {% block title %}PR #{{pr}}{% endblock %}
> % block header
>   <h1><a href="https://github.com/{{org}}/{{repo}}/pull/{{pr}}">{{org}}/{{repo}}#{{pr}}</a> Presubmit Test Results</h1>
> ```

Is "Presubmit" helpful here? The definition/usage of that varies-- sometimes, "presubmit" only refers to the tests triggered by the submit queue.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797)*

**Comment by fejta** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[gubernator/app.yaml, line 2 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUASU6KLfA7rojL2Ds:-KqUoArhJvNGM8yF8-Lu:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/261965b5015d11db73a1a9b5df95ca2077316982/gubernator/app.yaml#L2)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

why is this changing?
</blockquote></details>

Done.

---

*[gubernator/view_pr.py, line 34 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUAdl8LH-Y0bFbS5KS:-KqUloODtT267VNPPk_E:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr.py#L34)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

`for each job under a PR path`
</blockquote></details>

Done.

---

*[gubernator/view_pr.py, line 81 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUB6AOUsjI8tjvr33X:-KqUlqpylSKnh-9JiLdB:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr.py#L81)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

are these comments necessary?
</blockquote></details>

Done.

---

*[gubernator/view_pr.py, line 114 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUBSQ1AyjADpdn_ntr:-KqUm089sH2RQYyWZfOe:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr.py#L114)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

`'%s/%s' % (org, repo)` ?
</blockquote></details>

Done.

---

*[gubernator/view_pr_test.py, line 42 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqU9wRofPyHcBugkN5f:-KqUnLmT7NdwO5PTqIXI:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/view_pr_test.py#L42)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

Please match the format of the other Gubernator tests. See regex_test.py for an example of how this might look.
</blockquote></details>

Done.

---

*[gubernator/templates/pr.html, line 4 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797#-KqUCnvqUur5BqQJXspX:-KqUnaoMMlgvgHUjKdmA:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/37e7d479c5664994cd9153ccfd7e6c492332330c/gubernator/templates/pr.html#L4)):*
<details><summary><i>Previously, rmmh (Ryan Hitchman) wrote…</i></summary><blockquote>

Is "Presubmit" helpful here? The definition/usage of that varies-- sometimes, "presubmit" only refers to the tests triggered by the submit queue.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/3797)*

File: gubernator/templates/pr.html [hunk 1]
```diff
@@ -11,29 +10,29 @@
 
 % endif
 % if not rows
-	<h2>No Results</h2>
+  <h2>No Results</h2>
```
Comment (rmmh) [inline gubernator/templates/pr.html]: this will slow down page loads by microseconds!!

File: gubernator/view_pr_test.py [hunk 2]
```diff
@@ -16,18 +16,48 @@
 
 import datetime
 
-from webapp2_extras import securecookie
-
+# TODO(fejta): use non-relative imports
+# https://google.github.io/styleguide/pyguide.html?showone=Packages#Packages
```
Comment (rmmh) [inline gubernator/view_pr_test.py]: "All new code should import each module by its full package name."

no need to change!

**Comment by rmmh** [comment]
/lgtm

fix your test

File: gubernator/templates/pr.html (hunk 1 above)
Comment (fejta) [inline gubernator/templates/pr.html]: LOL because of the extra character?

**Pull request merged**